meson.build

```diff
@@ -0,0 +1,125 @@
+# Copyright © 2017 Dylan Baker
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+project(
+  'glew',
+  'c',
+  version : '2.1.0',
+  license: ['MIT', 'BSD-3'],
+  meson_version : '>= 0.40.0'
+)
+
+cc = meson.get_compiler('c')
+host_system = host_machine.system()
+
+c_args = [
+  '-DGLEW_NO_GLU',
+]
+ld_args = []
+glew_sources = [
+  'include/GL/wglew.h',
+  'include/GL/glew.h',
+  'include/GL/glxew.h',
+  'src/glew.c',
+]
+
+if get_option('default_library') == 'shared'
+  c_args += '-DGLEW_BUILD'
+else
+  c_args += '-DGLEW_STATIC'
+endif
+
+if host_system == 'windows'
+  dep_gl = cc.find_library('opengl32')
+else
+  dep_gl = dependency('gl')
+endif
+
+if host_system == 'windows'
+  windows = import('windows')
+
+  glew_sources += windows.compile_resources('build/glew.rc')
+  name = 'glew32'
+
+  if cc.get_id() == 'msvc'
+    c_args += [
+      '-DVC_EXTRALEAN',
+      '-GS-',
+    ]
+    ld_args += [
+      '-BASE:0x62AA0000',
+      '-nodefaultlib',
+      '-noentry',
+    ]
+  else
+    c_args += [
+      '-fno-builtin',
+      '-fno-stack-protector',
+    ]
+    ld_args += '-nostdlib'
+  endif
+else
+  name = 'GLEW'
+  potential_c_args = [
+    '-ansi',
+    '-Wall',
+    '-W',
+    '-Wcast-qual',
+    '-pedantic',
```

Comment:
Don't have `-Wall` and `-pedantic`. They come automatically based on warning levels. `-Wall` especially is something you can always assume to be there.